src/HIE/Bios/Cradle.hs

```diff
@@ -107,6 +109,7 @@ directCradle :: FilePath -> [String] -> Cradle
 directCradle wdir args =
   Cradle {
       cradleRootDir = wdir
+    , cradleDependencies = ["hie.yaml"] -- ^ Flags are supplied via 'hie.yaml'
```

Comment:
Actually, every cradle might need to monitor "hie.yaml" if it exists, since it can change the used cradle.